### ISSUE TITLE
Image should not take `bc.max()` when only width bounded.

### ISSUE THREAD

**Issue by JAicewizard** [linebender/druid#1184]
Currently the Image widget takes `bc.max`
This means what when you want to have images in a scroll-able horizontal list, the height of the images is `+inf `.
This means that the images are not displayed at all.
When the image is width bounded the image should take into consideration the width/height of the image itself and calculate the correct height from that.

The same should probably be done to height constrained image widgets.

**Comment by JAicewizard** [comment]
I could probably implement this, its not that hard.
If this is accepted I will start working on it.

**Comment by cmyr** [comment]
Without looking at it hard I 100% believe you're correct, `bc.max()` is rarely the correct constraint.

**Comment by JAicewizard** [comment]
I redid the calculations for the size, I tried making a test for this but the problem is that I need to set the size to +inf.
But the harness makes a creates an bitmap with the given size. Since we cannot make a bitmap with infinite size this obviously fails.
A possible solution would be to only create the bitmap after the layout has been done.

I will test this in an actual application, if that works I will create a PR.

I tried looking into the harness code, but it seems to be tightly integrated with the root window, so maybe that is left for someone who knows more about this.
